project_app/components/AdminLogin.js

```diff
@@ -0,0 +1,185 @@
+import { useState, useContext, useEffect } from "react";
+import { GlobalStateContext } from '../pages/_app';
+import styled from "styled-components";
+//import styles from "../styles/AdminLogin.module.css";
+import { CFormInput, CFormFloating, CFormLabel } from "@coreui/react";
+
+import "@coreui/coreui/dist/css/coreui.min.css";
+
+import Swal from "sweetalert2";
+
+import { useRouter } from 'next/router';
+
+export default function AdminLogin() {
+    const router = useRouter();
+    const { token, setToken } = useContext(GlobalStateContext);
+    const [emailOrUsername, setEmailOrUsername] = useState("");
+    const [password, setPassword] = useState("");
+
+    useEffect(() => {
+        console.log(token);
+        if (token != null) {
+            // Verify that token is not tampered with using api
+            router.push('adminpage');
+        }
+    }, [])
+
+    async function handleLoginClick() {
+        const response = await fetch(
+            `${process.env.NEXT_PUBLIC_API_URL}/api/admin/auth`,
+            {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json",
+                },
+                body: JSON.stringify({
+                    emailOrUsername: emailOrUsername,
+                    password: password,
+                }),
+            }
+        );
+
+
+        const resBody = await response.json();
+
+
+        if (!response.ok) {
+            Swal.fire({
+                title: resBody.heading,
+                text: resBody.message,
+                icon: "error",
+            });
+            return;
+        } else {
+            // Successfully logged in
+            //TO DO: Persist token between browser sessiosn. Send token through http only cookie to secure against xss, then use samesite strict to protect against csrf. this is instead of storing in local storage.
+            const tokenReceived = resBody.access;
+            setToken(tokenReceived);
+            router.push('/adminpage');
+
+        }
+
+    }
+
+    return (
+        <LoginLayout>
+            <AdminLoginTitle>Admin Login</AdminLoginTitle>
+
+            <AdminLoginDialog>
+                <DialogTitle>Login</DialogTitle>
+
+                <FormContainer>
+                    <DialogText>Welcome!</DialogText>
+
+                    <CFormFloating style={{ marginBottom: '1rem' }}>
+                        <CFormInput
+                            type="email"
+                            id="emailOrUsername"
+                            placeholder="name@example.com"
+                            value={emailOrUsername}
+                            onChange={(e) => {
+                                setEmailOrUsername(e.target.value);
+                            }}
+                            style={{ backgroundColor: "#2B2B2B", color: "#FFFFFF" }}
```

Comment:
can we have these styles in styledcomponent?

Reply:
This is what I actually wanted to do but I had problems with styling existing components, but a quick google search yielded results <3